app/decorators/odk/form_decorator.rb

```diff
@@ -63,8 +63,9 @@ def needs_last_saved_instance?
     # returns array of option sets that are referenced by the default dynamic calculation
     def referenced_option_sets(qing)
       codes = qing.default.scan(ODK::DynamicPatternParser::CODE_ONLY_REGEX)
-      questions = codes.flatten.map { |c| Question.with_code(c).first }
-      questions.any? ? questions.map { |q| q.option_set.presence } : []
+      mission = qing.mission
+      questions = codes.flatten.map { |c| Question.find_by(mission_id: mission.id, code: c) }
```

Comment:
Oh actually `Question.find_by` can return nil, which is causing spec `context "option sets for dynamic default answers"` to fail